webhooks/appmesh/mesh_validator.go

```diff
@@ -0,0 +1,64 @@
+package appmesh
+
+import (
+	"context"
+	appmesh "github.com/aws/aws-app-mesh-controller-for-k8s/apis/appmesh/v1beta2"
+	"github.com/aws/aws-app-mesh-controller-for-k8s/pkg/webhook"
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"reflect"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+	"strings"
+)
+
+const apiPathValidateAppMeshMesh = "/validate-appmesh-k8s-aws-v1beta2-mesh"
+
+// NewMeshValidator returns a validator for Mesh.
+func NewMeshValidator() *meshValidator {
+	return &meshValidator{}
+}
+
+var _ webhook.Validator = &meshValidator{}
+
+type meshValidator struct {
+}
+
+func (v *meshValidator) Prototype(req admission.Request) (runtime.Object, error) {
+	return &appmesh.Mesh{}, nil
+}
+
```

Comment:
Should we block a meshSpec selecting on system namespace ..say "kube-system", in their mesh spec as part of this?

Reply:
i don't think we should do that within code. customer should be have a controller-level flag to controller that if we decide to expose such functionality :D